src/qt/editor/welcomewidget.cpp

```diff
@@ -0,0 +1,507 @@
+
+    QSettings settings;
+    settings.beginGroup("InviwoMainWindow");
+
+    auto autoloadWorkspace =
+        new QCheckBox("&Automatically load most recently used workspace", rightColumn);
+    autoloadWorkspace->setChecked(settings.value("autoloadLastWorkspace", true).toBool());
+    QObject::connect(autoloadWorkspace, &QCheckBox::toggled, this, [](bool checked) {
+        QSettings settings;
+        settings.beginGroup("InviwoMainWindow");
+        settings.setValue("autoloadLastWorkspace", checked);
+        settings.endGroup();
+    });
+    verticalLayout_3->addWidget(autoloadWorkspace);
+
+    auto showOnStartup = new QCheckBox("&Show this page on startup", rightColumn);
+    showOnStartup->setChecked(settings.value("showWelcomePage", true).toBool());
+    QObject::connect(showOnStartup, &QCheckBox::toggled, this, [](bool checked) {
+        QSettings settings;
+        settings.beginGroup("InviwoMainWindow");
+        settings.setValue("showWelcomePage", checked);
+        settings.endGroup();
+    });
+    verticalLayout_3->addWidget(showOnStartup);
+
+    settings.endGroup();
+
+    gridLayout->addWidget(rightColumn, 0, 3, 3, 1);
+
+    // final layout adjustments
+    gridLayout->setRowStretch(1, 10);
+    gridLayout->setRowStretch(2, 1);
+    gridLayout->setColumnStretch(0, 1);
+    gridLayout->setColumnStretch(1, 3);
+    gridLayout->setColumnStretch(3, 2);
+
+    setLayout(gridLayout);
+
+    initChangelog();
+}
+
+void WelcomeWidget::updateRecentWorkspaces() {
+    filetree_->updateRecentWorkspaces(mainWindow_->getRecentWorkspaceList());
+}
+
+void WelcomeWidget::showEvent(QShowEvent *event) {
+    if (!event->spontaneous()) {
+        updateRecentWorkspaces();
+        filetree_->updateExampleEntries();
+    }
+    QWidget::showEvent(event);
+}
+
+void WelcomeWidget::keyPressEvent(QKeyEvent *event) {
+    if ((event->key() >= Qt::Key_0) && (event->key() <= Qt::Key_9)) {
+        int number = [key = event->key()]() {
+            if (key == Qt::Key_0) return 9;
+            return key - Qt::Key_1;
+        }();
+        if (filetree_->selectRecentWorkspace(number)) {
+            if ((event->modifiers() & Qt::ControlModifier) == Qt::ControlModifier) {
+                loadWorkspace_->animateClick();
+            }
+            event->accept();
+        }
+    } else if ((event->key() == Qt::Key_Return) || (event->key() == Qt::Key_Enter)) {
+        loadWorkspace_->animateClick();
+        event->accept();
+    }
+    QWidget::keyPressEvent(event);
+}
+
+void WelcomeWidget::loadWorkspace(const QString &filename, bool isExample) const {
+    bool controlPressed =
+        (mainWindow_->getInviwoApplicationQt()->keyboardModifiers() == Qt::ControlModifier);
+    if (mainWindow_->askToSaveWorkspaceChanges() &&
+        mainWindow_->openWorkspace(filename, isExample && !controlPressed)) {
+        mainWindow_->hideWelcomeScreen();
+    }
+}
+
+void WelcomeWidget::initChangelog() {
+    changelog_->setHtml(
```

Comment:
Lol

Reply:
Should still do the python thing here...

Reply:
Agreed. But maybe not as part of this PR?

Reply:
part of 5fbc0f6